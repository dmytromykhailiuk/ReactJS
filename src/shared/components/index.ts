export * from "./Footer";
export * from "./Logo";
export * from "./Button";
export * from "./Input";
export * from "./ThreeDotsIcon";
export * from "./ErrorBoundary";
