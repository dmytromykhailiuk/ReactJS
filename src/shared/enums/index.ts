export * from "./button-types";
export * from "./button-size";
export * from "./categories";
export * from "./sorting-options";
export * from "./movie-form-values";
export * from "./router-paths";
export * from "./modal-types";
