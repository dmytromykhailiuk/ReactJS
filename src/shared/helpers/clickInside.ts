import { SyntheticEvent } from "react";

export function isClickInside(
  event: SyntheticEvent,
  ...classes: string[]
): boolean {
  const { target, currentTarget } = event;

  function isInside(target: EventTarget & HTMLElement): boolean {
    if (target === currentTarget) {
      return true;
    }
    if (classes.some((className) => target.classList.contains(className))) {
      return false;
    }
    return isInside(target.parentElement);
  }
  return isInside(target as HTMLElement);
}
