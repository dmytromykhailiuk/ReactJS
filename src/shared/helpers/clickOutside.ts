export function clickOutside(className: string, fn: () => void) {
  function compere(target: HTMLElement) {
    if (!target || target.tagName === "BODY") {
      fn();
      return;
    }
    if (target.classList.contains(className)) {
      return;
    }
    compere(target.parentElement);
  }
  return (event: MouseEvent) => {
    compere(event.target as HTMLElement);
  };
}
