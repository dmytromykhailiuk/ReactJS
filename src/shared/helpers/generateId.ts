export function generateId() {
  return String(new Date().getTime());
}
