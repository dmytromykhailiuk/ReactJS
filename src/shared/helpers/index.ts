export * from "./filterMovies";
export * from "./sortMovies";
export * from "./generateId";
export * from "./clickOutside";
