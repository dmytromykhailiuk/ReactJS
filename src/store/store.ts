import { applyMiddleware, combineReducers, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';
import { spawn } from '@redux-saga/core/effects';
import { modalsReducer, ModalsState } from './modals';
import { moviesReducer, moviesSaga, MoviesState } from './movies';

function* rootSaga() {
  yield spawn(moviesSaga);
}

const sagaMiddleware = createSagaMiddleware();

export interface Store {
  movies: MoviesState;
  modals: ModalsState;
}

const store = createStore(
  combineReducers({ movies: moviesReducer, modals: modalsReducer }),
  composeWithDevTools(applyMiddleware(sagaMiddleware)),
);

sagaMiddleware.run(rootSaga);

export { store };
