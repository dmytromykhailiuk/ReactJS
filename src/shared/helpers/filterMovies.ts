import { Movie } from "models";
import { Categories } from "../enums";

export function filterMoviesBySearchingValue(
  movies: Movie[],
  searchingValue: string
) {
  if (!searchingValue) {
    return movies;
  }
  return movies.filter((movie) =>
    movie.title.toLowerCase().includes(searchingValue.toLowerCase())
  );
}

export function filterMoviesByCategory(movies: Movie[], categories: string) {
  if (categories === Categories.ALL) {
    return movies;
  }
  return movies.filter((movie) =>
    movie.genre.join(", ").toLowerCase().includes(categories.toLowerCase())
  );
}
