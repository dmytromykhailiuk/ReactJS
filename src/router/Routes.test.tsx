import { shallow, mount } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import { Router } from 'react-router';
import { createBrowserHistory } from 'history';
import { Provider } from 'react-redux';
import { Routes } from '.';
import { store } from '../store';
import { navigateToHome } from '../shared/helpers';

jest.mock('../shared/helpers');

describe('Routes', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Routes />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match snapshot with error page', () => {
    const history = {
      ...createBrowserHistory(),
      length: 1,
      location: {
        pathname: '/404',
        search: '',
        hash: '',
        state: undefined,
        key: '17mu3q',
      },
    } as any;
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <Routes />
        </Router>
      </Provider>,
    );
    wrapper.find('button').simulate('click');
    expect(navigateToHome as jest.Mock).toHaveBeenCalled();
  });
});
