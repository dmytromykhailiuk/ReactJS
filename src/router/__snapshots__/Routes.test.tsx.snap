// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes should match snapshot 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    path="/film/:id"
  />
  <Route
    component={[Function]}
    path="/search"
  />
  <Route
    path="/404"
    render={[Function]}
  />
  <Redirect
    to="/404"
  />
</Switch>
`;
